google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/scaladsl/GoogleBigQuerySource.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.bigquery.scaladsl
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{HttpMethods, HttpRequest}
+import akka.stream.Materializer
+import akka.stream.alpakka.googlecloud.bigquery.BigQueryConfig
+import akka.stream.alpakka.googlecloud.bigquery.client._
+import akka.stream.alpakka.googlecloud.bigquery.impl.parser.Parser.PagingInfo
+import akka.stream.alpakka.googlecloud.bigquery.impl.util.ConcatWithHeaders
+import akka.stream.alpakka.googlecloud.bigquery.impl.BigQueryStreamSource
+import akka.stream.scaladsl.{Sink, Source}
+import spray.json.JsObject
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+/**
+ * Scala API to create BigQuery sources.
+ */
+object GoogleBigQuerySource {
+
+  /**
+   * Read elements of `T` by executing HttpRequest upon BigQuery API.
+   */
+  def raw[T](
+      httpRequest: HttpRequest,
+      parserFn: JsObject => Try[T],
+      onFinishCallback: PagingInfo => NotUsed,
+      projectConfig: BigQueryConfig
+  )(implicit mat: Materializer, actorSystem: ActorSystem): Source[T, NotUsed] =
```

Comment:
Please wrap the source construction in `Source.setup` so that these factory methods don't require the implicit parameters.

Reply:
I was successful for the `BigQueryStreamSource` however I have difficulties not using implicits on certain cases where `runWith` is used
https://github.com/akka/alpakka/pull/2252/files/18dcb3f44f611c72ad05308744ebb39af58ce5c9#diff-c6cc45cda524e6037a1a94e3ca93c2c6R100

Reply:
I updated `raw` using `Source.setup` as an example.

Reply:
I changed the methods to use Source.setup. Also on utility methods like `listTables` and `listFields` the control for the Sink and collection is up to the user.